server/index.js

```diff
@@ -9,19 +24,52 @@ app.use(express.static('public'))
 
 app.use(bodyParser.json())
 
-const privateKey = '0x3daa79a26454a5528a3523f9e6345efdbd636e63f8c24a835204e6ccb5c88f9e'
+const exchangePublicToken = async token => {
+  const tokenResponse = await plaidClient.exchangePublicToken(token)
+  if (tokenResponse.status_code !== 200) {
+    logger.error({ status: tokenResponse.status_code }, 'Error exchanging public token')
+    throw Error(`[exchangePublicToken] ${tokenResponse.error_code}: {tokenResponse.error_message}`)
+  }
+  return tokenResponse
+}
+
+const getBankAccountData = async accessToken => {
+  const response = await plaidClient.getAuth(accessToken)
+  if (response.status_code !== 200) {
+    logger.error({ status: response.status_code }, 'Error getting bank account')
+    throw Error(`[getBankAccountData] ${response.error_code}: {response.error_message}`)
+  }
+  return response
+}
 
 app.post('/api/get-tx-data', (req, res) => {
-  const { ethAccount } = req.body
-  const bankAccount = 'MyAccount-' + Math.floor(1000 * Math.random())
+  const { ethAccount, token } = req.body
+
+  logger.info({ ethAccount }, 'Getting bank accounts')
+
+  return exchangePublicToken(token)
+    .then(tokenResponse => {
+      const accessToken = tokenResponse.access_token
+      return getBankAccountData(accessToken)
+    })
+    .then(accountData => {
+      logger.info({ accountData }, 'Got account data')
+      const { numbers } = accountData
+      const bankAccount = numbers.ach[0].account
```

Comment:
This `ach` property still doesn't work for me (`numbers` is just an array in my log). I guess it works on your side. Any idea why are we getting different results? Maybe different versions of the API?

Reply:
```
poa-poba on  feature/server-functions [$]  ⬢ v8.9.4 took 5s
🦄 npm -v plaid
5.8.0
```

and this is the structure of `numbers`
```json
numbers: { ach:
    [ { account: '1111222233330000',
        account_id: '76o8QJjk1es3yrlzkBAZUJEPQxgZwxFNvG1xW',
        routing: '011401533',
        wire_routing: '021000021' },
      { account: '1111222233331111',
        account_id: 'edLbaqrnjzUepmr6BqPys4e1wXyRGXiM6wVmg',
        routing: '011401533',
        wire_routing: '021000021' } ],
   eft: [] 
}
```

Reply:
from Google:
_[...] ACH stands for the Automated Clearing House and is basically the process of moving funds electronically between bank accounts. ACH processing can also be called EFT (electronic funds transfer) [...]_